app/frontend/state.ts

```diff
@@ -212,42 +237,78 @@ const initialState: State = {
     ADALITE_CONFIG.ADALITE_ENV === 'local' && ADALITE_CONFIG.ADALITE_DEVEL_AUTO_LOGIN === 'true',
 
   // shelley
-  displayStakingPage: false,
+  selectedMainTab: 'Sending',
   shelleyDelegation: {
     delegationFee: 0.0,
     selectedPool: {
       poolHash: '',
     },
   },
   gettingPoolInfo: false,
-  shelleyBalances: {
-    nonStakingBalance: 0,
-    stakingBalance: 0,
-    rewardsAccountBalance: 0,
-  },
-  shelleyAccountInfo: {
-    accountPubkeyHex: '',
-    currentEpoch: 0,
-    delegation: {},
-    hasStakingKey: false,
-    rewards: 0,
-    rewardDetails: {
-      upcoming: null,
-      nearest: null,
-      currentDelegation: null,
-    },
-    value: 0,
-  },
   txConfirmType: '',
   txSuccessTab: '',
   keepConfirmationDialogOpen: false,
-  poolRecommendation: {
-    isInRecommendedPoolSet: true,
-    recommendedPoolHash: '',
-    status: '',
-    shouldShowSaturatedBanner: false,
-  },
   isBigDelegator: false,
+  accountsInfo: [
+    {
+      balance: 0,
+      shelleyBalances: {
+        stakingBalance: 0,
+        nonStakingBalance: 0,
+        rewardsAccountBalance: 0,
+      },
+      stakePubkeyHex: '',
+      shelleyAccountInfo: {
+        accountPubkeyHex: '',
+        shelleyXpub: '',
+        byronXpub: '',
+        stakingKey: null,
+        stakingAccountAddress: '',
+        currentEpoch: 0,
+        delegation: {},
+        hasStakingKey: false,
+        rewards: 0,
+        rewardDetails: {
+          upcoming: null,
+          nearest: null,
+          currentDelegation: null,
+        },
+        value: 0,
+      },
+      transactionHistory: [],
+      stakingHistory: [],
+      visibleAddresses: [],
+      poolRecommendation: {
+        isInRecommendedPoolSet: true,
+        recommendedPoolHash: '',
+        status: '',
+        shouldShowSaturatedBanner: false,
+      },
+      isUsed: false,
+      accountIndex: 0,
+    },
+  ],
+  shouldNumberAccountsFromOne: false,
+  sourceAccountIndex: 0,
+  activeAccountIndex: 0,
+  targetAccountIndex: 0,
+  totalWalletBalance: 0,
+  totalRewardsBalance: 0,
+  shouldShowSendTransactionModal: false,
+  shouldShowDelegationModal: false,
+  sendTransactionTitle: '',
+  delegationTitle: '',
+  poolCertTxVars: {
+    shouldShowPoolCertSignModal: false,
+    ttl: 0,
+    signature: null,
+    plan: null,
+  },
 }
+export type SetStateFn = (newState: Partial<State>) => void
+export type GetStateFn = () => State
+
+export const sourceAccountState = (state: State) => state.accountsInfo[state.sourceAccountIndex]
```

Comment:
isn't that more like `sourceAccountInfo`? and probably `getSourceAccountInfo` would be a better fit as it's a function.

Reply:
You are right, I ll fix it :+1: